op/opstatus.go

```diff
@@ -45,7 +50,13 @@ func (s *Status) SetRevision(revision string) *Status {
 // Multiple checkers are allowed.  The checker functions should be capable of
 // being called concurrently (with each other and with themselves).
 func (s *Status) AddChecker(name string, checkerFunc func(cr *CheckResponse)) *Status {
```

Comment:
I like the goal of allowing replacement of checkers, but this changes the semantics such that it's no longer accurate to say that you are "adding" a checker.  To me, it's not so much about whether anyone ever adds two with the same name, but that there's an implied replace here that contradicts the method name (and the documentation).  Changing the documentation is a possibility so it matches the behaviour, but then both still don't  fit with the method name.
Since it's probably rare to replace, how about we leave the `AddChecker` as it is, and introduce a new `RemoveCheckers` that removes all checkers with a given name?  That would leave all existing code alone, but still enable this use case.

Reply:
Sounds sensible, I'll close this down & reopen with a reworked PR

Reply:
Have opened https://github.com/utilitywarehouse/go-operational/pull/10